src/v/storage/compaction_reducers.cc

```diff
@@ -269,6 +269,7 @@ ss::future<ss::stop_iteration> copy_data_segment_reducer::do_compaction(
           batch.last_offset(),
           batch.header().first_timestamp,
           batch.header().max_timestamp,
+          std::nullopt,
```

Comment:
Why do we drop the broker timestamp when doing compaction? It's a bit weird that compacted segments have different behaviour.

Reply:
forgot to include the part where i set broker_timestamp outside this function. it's in segment_utils::do_self_compact https://github.com/redpanda-data/redpanda/pull/12991/files#diff-c79a67494746e1dac357cbfc721f6718275cc322e24c34dfc8f8f6cdcae546d9R513-R549